code/network/multimsgs.cpp

```diff
@@ -8011,14 +8011,20 @@ static constexpr size_t animation_forced_bit = 1 << 1;
 static constexpr size_t animation_instant_bit = 1 << 2;
 static constexpr size_t animation_pause_bit = 1 << 3;
 
-void send_animation_triggered_packet(int animationId, int pmi, const animation::ModelAnimationDirection& direction, bool force, bool instant, bool pause, const int* /*time*/) {
+void send_animation_triggered_packet(unsigned int animationId, object* parent_object, ushort special_mode, const animation::ModelAnimationDirection& direction, bool force, bool instant, bool pause, const int* /*time*/) {
 	int packet_size;
+	ushort netsig_to_send = 0;
 	ubyte data[MAX_PACKET_SIZE];
 
+	Assertion(special_mode != 0 || parent_object != nullptr, "Tried to synchronize an animation with neither an attached object, nor a valid special object mode.");
+	if (special_mode == 0)
+		netsig_to_send = parent_object->net_signature;
```

Comment:
Don't forget the nullptr check on `parent_object` here since release builds would just crash without warning. It would be best to just `return` if neither option is valid and not send the packet.